db_test.go

```diff
@@ -97,28 +91,30 @@ func (s *ReformSuite) TestAbortedTransaction() {
 	s.Require().NoError(s.tx.Rollback())
 	s.q = nil
 
-	setIdentityInsert(s.T(), DB.Querier, "people", true)
-
 	person1 := &Person{ID: 42, Email: pointer.ToString(faker.Internet().Email())}
 	person2 := &Person{ID: 43, Email: pointer.ToString(faker.Internet().Email())}
 
 	// commit fails
 	tx, err := DB.Begin()
 	s.Require().NoError(err)
-	s.NoError(tx.Insert(person1))
-	s.EqualError(tx.Insert(person1), `pq: duplicate key value violates unique constraint "people_pkey"`)
-	s.EqualError(tx.Insert(person2), `pq: current transaction is aborted, commands ignored until end of transaction block`)
-	s.EqualError(tx.Commit(), `pq: Could not complete operation in a failed transaction`)
+	s.NoError(insertPersonWithID(s.T(), tx.Querier, person1))
+	s.Contains(insertPersonWithID(s.T(), tx.Querier, person1).Error(), `duplicate key value violates unique constraint "people_pkey"`)
+	s.Contains(insertPersonWithID(s.T(), tx.Querier, person2).Error(), `current transaction is aborted, commands ignored until end of transaction block`)
+	err = tx.Commit()
+	s.Require().Error(err)
+	if err.Error() != `pq: Could not complete operation in a failed transaction` && err.Error() != `commit unexpectedly resulted in rollback` {
+		s.Failf("unexpected error", "actual: %q", err)
+	}
```

Comment:
My [original changeset](https://github.com/AlekseyMartynov/reform/commit/bea984fbf30c7452c7ebbe282ac64e6d015f3363) was based on the `mssql-update` branch. So was the original PR (according to [AppVeyor logs](https://ci.appveyor.com/project/AlekSi/reform/build/474)). As I understand, later you've changed the PR to target `v1-3` and something went wrong.

Now, in my opinion, it would be cleaner to cherry-pick the original commit on top of the current `v1-3`. There'll be a conflict revealing the source of the unwanted change:

```go
<<<<<<< HEAD
	s.NoError(tx.Insert(person1))
	s.EqualError(tx.Insert(person1), `pq: duplicate key value violates unique constraint "people_pkey"`)
	s.EqualError(tx.Insert(person2), `pq: current transaction is aborted, commands ignored until end of transaction block`)
	s.EqualError(tx.Commit(), `pq: Could not complete operation in a failed transaction`)
=======
	s.NoError(insertPersonWithID(s, tx.Querier, person1))
	s.Contains(insertPersonWithID(s, tx.Querier, person1).Error(), `duplicate key value violates unique constraint "people_pkey"`)
	s.Contains(insertPersonWithID(s, tx.Querier, person2).Error(), `current transaction is aborted, commands ignored until end of transaction block`)
	err = tx.Commit()
	s.Require().Error(err)
	if err.Error() != `pq: Could not complete operation in a failed transaction` && err.Error() != `commit unexpectedly resulted in rollback` {
		s.Failf("unexpected error", "actual: %q", err)
	}
>>>>>>> bea984f... Rework SET IDENTITY_INSERT for Go 1.10
```

Reply:
![mr-holmes](https://user-images.githubusercontent.com/4426185/41668554-458d6164-74b8-11e8-81b6-a96f35732f30.png)